sled-agent/src/bootstrap/http_entrypoints.rs

```diff
@@ -168,3 +169,23 @@ async fn sled_reset(
     ba.sled_reset().await.map_err(|e| Error::from(e))?;
     Ok(HttpResponseUpdatedNoContent())
 }
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct SledShutdownRequest {
+    /// The reason for the shutdown.
```

Comment:
Smart to include this. I know the number of callers here is currently "zero", but we'll want this soon, I bet.